hoomd/data/__init__.py

```diff
@@ -1,7 +1,14 @@
 # Copyright (c) 2009-2022 The Regents of the University of Michigan.
 # Part of HOOMD-blue, released under the BSD 3-Clause License.
 
-"""Particle data local access."""
+"""Access State data on the local rank.
```

Comment:
This issue occurs elsewhere too. It happens because @b-butler placed some classes, like `TypeParameter` in other modules and imports them into the canonical namespace location. I have not yet found a way to tell sphinx to name these with their canonical name while also using `autodoc`.

Reply:
Actually, the issue with TypeParameter is that it is not imported into the top level `data`. Should it be @b-butler ? I find it odd to publish an API for a public class where we do not publish the docs for the module it is in.

The issue I mentioned for the other classes not using canonical names is in the "Bases:" output. I'm testing a workaround for that now.

Reply:
Unfortunately, if we do this, I get 
```
WARNING: more than one target found for 'any' cross-reference 'TypeParameter': could be :py:class:`hoomd.data.TypeParameter` or :py:class:`hoomd.data.typeparam.TypeParameter
```
in many places throughout the codebase. This isn't trivial to fix because many of the places we reference `TypeParameter` in property types which are very constrained on line length. I don't think it is worth the time to reformat that much documentation.